### PR TITLE
fix(kafka): Add Kraft to Kafka containers

### PR DESCRIPTION
**Pull request by jfmlima**
Following a similar strategy as several other testcontainers implementations, this PR introduces the possibility to run Kafka in KRAft mode.

```py
with KafkaContainer().with_kraft() as container:
    # Test something with/on KRaft mode
```

### CODE REVIEW

File: modules/kafka/testcontainers/kafka/__init__.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 import time
+import uuid
 from io import BytesIO
 from textwrap import dedent
 
 from testcontainers.core.container import DockerContainer
 from testcontainers.core.utils import raise_for_deprecated_parameter
+from testcontainers.core.version import ComparableVersion
 from testcontainers.core.waiting_utils import wait_for_logs
-from testcontainers.kafka._redpanda import RedpandaContainer
```
Comment (alexanderankin) [inline modules/kafka/testcontainers/kafka/__init__.py:9]: ?

**Review by alexanderankin — CHANGES_REQUESTED** [review]
if you can make it pass, ill merge.

lmk if you need any help running the ci locally

File: modules/kafka/testcontainers/kafka/__init__.py (hunk 1 above)
Comment (jfmlima) [reply modules/kafka/testcontainers/kafka/__init__.py:9]: sorry about that, it was my IDE auto-removing unused imports, already fixed it 👍

**Comment by jfmlima** [comment]
> if you can make it pass, ill merge.
> 
> lmk if you need any help running the ci locally

Thanks, I'm using act and see how it goes, hopefully I've fixed all issues, but running locally now just in case 👍

**Comment by jfmlima** [comment]
> if you can make it pass, ill merge.
> 
> lmk if you need any help running the ci locally

Hey @alexanderankin do you mind giving me some tips on how to run the ci locally please? act alone was not enough as it has some issues with the python setup. Thanks!

**Comment by jfmlima** [comment]
@alexanderankin all sorted, feel free to re-run the ci 👍

**Comment by alexanderankin** [comment]
sorry for the delay in response, looks like you got it sorted. here it is anyway

```
python -m venv ~/.poetry-venv && source ~/.poetry-venv/bin/activate && pip install poetry && ln -s $(which poetry) ~/.local/bin # or wherever you put bins
deactivate

git clone.... && cd testcontainers-python

poetry install --with dev -E kafka
poetry shell # starts shell

make modules/kafka/tests
make modules/kafka/<tab to autocomplete>

or even just
pytest -s modules/kafka/tests

and then also remember to
pre-commit run -a
```

**Review by alexanderankin — APPROVED** [review]

**Pull request merged**